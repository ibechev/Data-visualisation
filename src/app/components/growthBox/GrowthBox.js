import React from "react";

const GrowthBox = ({ prevValue, currentValue }) => {
  console.log("prevValue:", prevValue);
  console.log("currentValue:", currentValue);

  const label =
    currentValue > prevValue
      ? "Increase"
      : prevValue === currentValue
      ? "Unchanged"
      : "Decrease";

  const iconClass =
    currentValue > prevValue
      ? "fas fa-angle-double-up"
      : prevValue === currentValue
      ? "fas fa-grip-lines"
      : "fas fa-angle-double-down";

  return (
    <div className="growth-box">
      <div className="growth-box-wrapper">
        <i className={iconClass} />

        <span className="growth-number">{currentValue - prevValue}</span>
      </div>
      <span className="growth-box-label">{label}</span>
    </div>
  );
};

export default GrowthBox;
