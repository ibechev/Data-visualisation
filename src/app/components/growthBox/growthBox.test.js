import React from "react";
import { shallow } from "enzyme";
import GrowthBox from "./GrowthBox";
import renderer from "react-test-renderer";
import { propTypeErrors } from "../../../../tests/config/testUtils";

describe("Expect <GrowthBox />", () => {
  const defaultProps = {
    prevValue: 100,
    currentValue: 150
  };
  const INCREASE = "Increase";
  const DECREASE = "Decrease";
  const UNCHANGED = "Unchanged";

  it("to render correctly", () => {
    const wrapper = renderer.create(<GrowthBox {...defaultProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to not throw error with correct props", () => {
    propTypeErrors(<GrowthBox {...defaultProps} />, defaultProps);
  });

  it("to set the correct label, icon and number when prop 'prevValue' is less than prop 'currentValue'", () => {
    const wrapper = shallow(<GrowthBox {...defaultProps} />);

    expect(wrapper.find("span.growth-box-label").text()).toEqual(INCREASE);
    expect(wrapper.find("i.fa-angle-double-up")).toHaveLength(1);
    expect(wrapper.find("span.growth-number").text()).toEqual(
      (defaultProps.currentValue - defaultProps.prevValue).toString()
    );
  });

  it("to set the correct label, icon and number when prop 'prevValue' is greater than prop 'currentValue'", () => {
    const prevValue = 150;
    const currentValue = 100;

    const wrapper = shallow(
      <GrowthBox prevValue={prevValue} currentValue={currentValue} />
    );

    expect(wrapper.find("span.growth-box-label").text()).toEqual(DECREASE);
    expect(wrapper.find("i.fa-angle-double-down")).toHaveLength(1);
    expect(wrapper.find("span.growth-number").text()).toEqual(
      (currentValue - prevValue).toString()
    );
  });

  it("to set the correct label, icon and number when prop 'prevValue' is equal to prop 'currentValue'", () => {
    const prevValue = 150;
    const currentValue = 150;

    const wrapper = shallow(
      <GrowthBox prevValue={prevValue} currentValue={currentValue} />
    );

    expect(wrapper.find("span.growth-box-label").text()).toEqual(UNCHANGED);
    expect(wrapper.find("i.fa-grip-lines")).toHaveLength(1);
    expect(wrapper.find("span.growth-number").text()).toEqual(
      (currentValue - prevValue).toString()
    );
  });
});
