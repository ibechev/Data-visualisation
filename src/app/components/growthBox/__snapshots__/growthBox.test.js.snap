// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <GrowthBox /> to render correctly 1`] = `
<div
  className="growth-box"
>
  <div
    className="growth-box-wrapper"
  >
    <i
      className="fas fa-angle-double-up"
    />
    <span
      className="growth-number"
    >
      50
    </span>
  </div>
  <span
    className="growth-box-label"
  >
    Increase
  </span>
</div>
`;
